cmd/memberagent/main.go

```diff
@@ -367,6 +369,13 @@ func Start(ctx context.Context, hubCfg, memberConfig *rest.Config, hubOpts, memb
 			klog.V(2).Info("setting up the Azure property provider")
 			// Note that the property provider, though initialized here, is not started until
 			// the specific instance wins the leader election.
+			cloudConfig, err := azureCloudConfig.NewCloudConfigFromFile(*cloudConfigFile)
```

Comment:
could you please remove this or add a TODO here as the cloudConfig is not used?

I just notice that that our e2e tests are using property-provider=azure. Need to think about how to test your feature.

If you want to test the feature using this e2e tests, we figure out a way to setup the azure resource correctly. Otherwise, your init azure client call will fail and break the e2e tests.